core/blockchain.go

```diff
@@ -1989,6 +1996,21 @@ func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals bool) (int, er
 	return it.index, err
 }
 
+func (bc *BlockChain) updateHighestVerifiedHeader(header *types.Header) {
+	if header == nil || header.Number == nil {
+		return
+	}
+	currentHeader := bc.highestVerifiedHeader.Load().(*types.Header)
+	if currentHeader == nil || currentHeader.Number == nil || currentHeader.Number.Cmp(header.Number) < 0 {
```

Comment:
when we talk about `highest`, it is not meaning the height, it means the total difficulty. So I will suggest  to compare: `h.chain.GetTd(parenthash, parentNumber) + header.Difficulty`